dask/dataframe/shuffle.py

```diff
@@ -545,15 +565,19 @@ def collect(p, part, meta, barrier_token):
 
 
 def set_partitions_pre(s, divisions):
-    partitions = pd.Series(divisions).searchsorted(s, side="right") - 1
-    partitions[(s >= divisions[-1]).values] = len(divisions) - 2
+    if isinstance(divisions, list) or isinstance(divisions, tuple):
```

Comment:
I'd like to remove these type checks if we can, but, while they're here, you may be interested in knowing that this is often written as 

```python
if isnstance(divisions, (list, tuple)):
```

Reply:
Note that I am no longer doing this same kind of type checking here, but am still checking the type of df in other places (`set_index` and `rearrange_by_divisions`) to define `meta` and `divisions` as appropriate series objects.